tests/load/locustfiles/load.py

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Load test shape module."""
+import math
+from typing import Type
+
+import numpy
+from locust import LoadTestShape, User
+from locustfile import AutopushUser
+
+TickTuple = tuple[int, float, list[Type[User]]]
+
+
+class QuadraticTrend:
+    """A class that defines a quadratic LoadTestShape trend."""
+
+    a: float
+    b: float
+    c: float
+
+    def __init__(self, max_run_time: int, max_users: int):
+        self.a, self.b, self.c = numpy.polyfit(
+            [0, (max_run_time / 2), max_run_time], [0, max_users, 0], 2
+        )
+
+    def calculate_users(self, run_time: int) -> int:
+        """Determined the number of active users given a run time.
+
+        Returns:
+            int: The number of users
+        """
+        return int(round((self.a * math.pow(run_time, 2)) + (self.b * run_time) + self.c))
+
+
+class AutopushLoadTestShape(LoadTestShape):
+    """A load test shape class for Autopush (Duration: 10 minutes, Users: 83300).
+    Note: The Shape class assumes that the workers can support the generated spawn rates.
+    """
+
+    MAX_RUN_TIME: int = 600  # 10 minutes
+    MAX_USERS: int = 83300
```

Comment:
This feels a bit like a magic number. Can you add a comment as to why you picked this or if it's just an arbitrary amount?

Reply:
It is indeed a bit magic like, I share your sentiment. Comments are added, what do you think?

Reply:
If someone else six months from now decides to change this value to `79481` (because those were his lucky lottery numbers), what impact would that have?

Reply:
The max users can be seen as the peak of the parabola created by the trend. Is that helpful information to write down? 

A little bit I would have to trust that whoever maintains these numbers reads a bit about Locust. No matter what you are asked to input a 'Magic' user number.

Reply:
Ah, Yes! that would be helpful. Maybe we rename that as `peak_user_load` and add a note that the app will ramp up and down to that number, and that for now the number is arbitrarily chosen. 

Sorry to be a pain about all of this, but I've been caught by things like that in the past and I've learned to treat myself as the idiot that I will become.